tests/posix/barriers/src/main.c

```diff
@@ -10,10 +10,20 @@
 #include <zephyr/sys/util.h>
 #include <zephyr/ztest.h>
 
+#if defined(CONFIG_SPARC)
+/*
+ * qemu_leon3 asserts when the attribute is not static. No idea why.
```

Comment:
This looks like something which should be debugged and probably fixed.
(Needing to make a stack variable persistent tends to be the result of a pointer to that variable being used later than expected. Which may work in some cases out of luck).

Reply:
Feel free to debug. I attempted to do so, but `qemu_leon` simply crashes. Fixing it is very much out of the scope of this PR.

Also, I may be generalizing, but It's fairly unlikely that we will see SPARC based products running Zephyr anytime soon. `qemu_leon3` is the only platform that exhibits this behaviour. I'm comfortable leaving the workaround in-tree until there is a pressing reason to fix it.

Reply:
@aescolar - fixed a long time ago already ;-)

See the linked issue (already applied to this PR)
https://github.com/zephyrproject-rtos/zephyr/issues/98258

Reply:
I'll mark this as resolved (since it is)

Reply:
Thanks @kartben - without actually mentioning what he wanted changed, I was not really sure what @aescolar meant. It's difficult to just know.

Reply:
@kartben - that's why I didn't mark it as resolved :-)

On the contrary, @aescolar - please mark your comment as resolved, since it was addressed 4 days ago.